src/framework/xr/xr-mesh.js

```diff
@@ -0,0 +1,179 @@
+import { EventHandler } from "../../core/event-handler.js";
+import { Vec3 } from "../../core/math/vec3.js";
+import { Quat } from "../../core/math/quat.js";
+
+let ids = 0;
+
+/**
+ * Detected Mesh instance that provides its transform (position, rotation),
+ * triangles (vertices, indices) and its semantic label. Any of its properties can
+ * change during its lifetime.
+ *
+ * @category XR
+ */
+class XrMesh extends EventHandler {
+    /**
+     * @type {number}
+     * @private
+     */
+    _id;
+
+    /**
+     * @type {import('./xr-mesh-detection.js').XrMeshDetection}
+     * @private
+     */
+    _meshes;
+
+    /**
+     * @type {XRMesh}
+     * @private
+     */
+    _xrMesh;
+
+    /**
+     * @type {number}
+     * @private
+     */
+    _lastChanged = 0;
+
+    /**
+     * @type {Vec3}
+     * @private
+     */
+    _position = new Vec3();
+
+    /**
+     * @type {Quat}
+     * @private
+     */
+    _rotation = new Quat();
+
+    /**
+     * Create a new XrMesh instance.
+     *
+     * @param {import('./xr-mesh-detection.js').XrMeshDetection} meshes - Mesh Detection
+     * interface.
+     * @param {XRMesh} xrMesh - XRMesh that is instantiated by WebXR system.
+     * @hideconstructor
+     */
+    constructor(meshes, xrMesh) {
+        super();
+
+        this._id = ++ids;
+        this._meshes = meshes;
+        this._xrMesh = xrMesh;
+        this._lastChanged = this._xrMesh.lastChangedTime;
+    }
+
+    /**
+     * Fired when {@link XrMesh} is removed.
+     *
+     * @event XrMesh#remove
+     * @example
+     * mesh.once('remove', function () {
+     *     // mesh is no longer available
+     * });
+     */
+
+    /**
+     * Fired when {@link XrMesh} attributes such as vertices, indices and/or label have been changed.
+     * Position and rotation can change at any time without triggering a `change` event.
+     *
+     * @event XrMesh#change
+     * @example
+     * mesh.on('change', function () {
+     *     // mesh attributes have been changed
+     * });
+     */
+
+    /**
+     * Unique identifier of a mesh.
+     *
+     * @type {number}
+     */
+    get id() {
```

Comment:
any reason this is exposed?
It implies that perhaps if the same mesh gets added later after it was removed, it would have the same Id.

Reply:
It is for convenience, and consistency with XrPlanes. As XrMesh can be added/removed during the session, and developers might use ID's for their indexes and identification. Perhaps with common use of Map and being able to use object as a key, instead of old-school object<id, XrMesh> type index, this is not needed anymore.